### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      .then(() => jwt.verify(req.token, settings.jwt.secret))
+      .then(model => {
+        req.model = {
+          ...model,
+          id: 'register-user'
+        };
+        next();
+      })
+      .catch(next);
+  });
```
Comment (lennym) [inline lib/app.js]: I'd probably just do all this lot in the `param` method.

File: lib/pages/register/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      if (req.existingUser) {
+        req.form.schema = {};
+        res.template = existingTemplate.default || existingTemplate;
+      }
+      next();
+    }
+  }));
+
+  app.use((req, res, next) => {
+    res.locals.static.schema = req.form.schema;
```
Comment (lennym) [inline lib/pages/register/index.js]: This feels like a thing that might be best done in the form router.

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.drone.yml
+deploy
+node_modules
```
Comment (lennym) [inline .dockerignore]: This is a very good point. I should remind myself to check we're doing it everywhere else too.

File: lib/pages/register/index.js (hunk 2 above)
Comment (joefitter) [reply lib/pages/register/index.js]: Ah yeah, I actually fixed it there, I wrote this before when I was using a previous version of pages - will remove

File: lib/app.js (hunk 1 above)
Comment (joefitter) [reply lib/app.js]: 👍

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    image: node:10
+    secrets:
+      - npm_auth_token
+    commands:
+      - npm ci
+      - npm test
+    when:
+      event: [push, pull_request, tag]
+
+  compile:
```
Comment (lennym) [inline .drone.yml]: It's be good to get an [audit step](https://github.com/UKHomeOffice/asl/blob/master/.drone.yml#L13-L20) in here too, but not critical at this point.

File: .kube.dev.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      secret: true
+      name: asl-dev-session
+      key: secret
+    KEYCLOAK_REALM: Asl-dev
+    KEYCLOAK_URL: https://sso-dev.notprod.homeoffice.gov.uk/auth
+    KEYCLOAK_CLIENT: asl-dev-connect
+    KEYCLOAK_SECRET:
+      secret: true
+      name: asl-dev-auth
+      key: secret
```
Comment (lennym) [inline .kube.dev.yaml]: It feels like there's a lot more environment variables being used in config.js than are being defined here.

**Pull request merged**